internal/component/database_observability/mysql/collector/query_sample.go

```diff
@@ -106,34 +107,52 @@ func (c *QuerySample) fetchQuerySamples(ctx context.Context) error {
 			level.Error(c.logger).Log("msg", "failed to iterate rs", "err", err)
 			break
 		}
-		var digest, query_sample_text, query_sample_seen, query_sample_timer_wait string
-		err := rs.Scan(&digest, &query_sample_text, &query_sample_seen, &query_sample_timer_wait)
+
+		var digest, sampleText, sampleSeen, sampleTimerWait string
+		err := rs.Scan(&digest, &sampleText, &sampleSeen, &sampleTimerWait)
 		if err != nil {
-			level.Error(c.logger).Log("msg", "failed to scan query samples", "digest", digest, "err", err)
-			break
+			level.Error(c.logger).Log("msg", "failed to scan result set", "err", err)
```

Comment:
We have this error message in a number of places where we do row scanning. Having the error message indicate which place it's happening (scanning query sample) is useful to keep in

Reply:
Reworded the message.